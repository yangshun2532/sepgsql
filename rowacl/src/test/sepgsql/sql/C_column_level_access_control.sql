--
-- testcases of column level access control
--
-- selinux_user: unconfined_u
-- selinux_role: unconfined_r
-- selinux_type: sepgsql_test_t
-- selinux_range: s0

SELECT * FROM c1;	-- to be failed

SELECT b, c FROM c1;

SELECT b, c FROM c1 WHERE a > 10;	-- to be failed

UPDATE c1 SET b = 'mokeke';		-- to be failed

SELECT * FROM c2;	-- to be failed

SELECT b, d FROM c2;

INSERT INTO c2 (a, b, d) VALUES ('aaa', 'bbb', 'ddd');	-- to be failed

INSERT INTO c2 (a, d) VALUES ('aaa', 'ddd');

UPDATE c2 SET a = a || '_update', d = d || '_update';	-- to be failed

DELETE FROM c2;
