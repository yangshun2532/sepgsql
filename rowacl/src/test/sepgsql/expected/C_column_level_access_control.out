--
-- testcases of column level access control
--
-- selinux_user: unconfined_u
-- selinux_role: unconfined_r
-- selinux_type: sepgsql_test_t
-- selinux_range: s0
SELECT * FROM c1;	-- to be failed
psql:./C_column_level_access_control.sql:9: ERROR:  SELinux: denied { select } scontext=unconfined_u:unconfined_r:sepgsql_test_t:s0 tcontext=system_u:object_r:sepgsql_secret_table_t:s0 tclass=db_column name=c1.a
SELECT b, c FROM c1;
 b | c 
---+---
(0 rows)

SELECT b, c FROM c1 WHERE a > 10;	-- to be failed
psql:./C_column_level_access_control.sql:13: ERROR:  SELinux: denied { use } scontext=unconfined_u:unconfined_r:sepgsql_test_t:s0 tcontext=system_u:object_r:sepgsql_secret_table_t:s0 tclass=db_column name=c1.a
UPDATE c1 SET b = 'mokeke';		-- to be failed
psql:./C_column_level_access_control.sql:15: ERROR:  SELinux: denied { update } scontext=unconfined_u:unconfined_r:sepgsql_test_t:s0 tcontext=system_u:object_r:sepgsql_ro_table_t:s0 tclass=db_table name=c1
SELECT * FROM c2;	-- to be failed
psql:./C_column_level_access_control.sql:17: ERROR:  SELinux: denied { select } scontext=unconfined_u:unconfined_r:sepgsql_test_t:s0 tcontext=system_u:object_r:sepgsql_secret_table_t:s0 tclass=db_column name=c2.c
SELECT b, d FROM c2;
 b | d 
---+---
(0 rows)

INSERT INTO c2 (a, b, d) VALUES ('aaa', 'bbb', 'ddd');	-- to be failed
psql:./C_column_level_access_control.sql:21: ERROR:  SELinux: denied { insert } scontext=unconfined_u:unconfined_r:sepgsql_test_t:s0 tcontext=system_u:object_r:sepgsql_ro_table_t:s0 tclass=db_column name=c2.b
INSERT INTO c2 (a, d) VALUES ('aaa', 'ddd');
UPDATE c2 SET a = a || '_update', d = d || '_update';	-- to be failed
psql:./C_column_level_access_control.sql:25: ERROR:  SELinux: denied { update } scontext=unconfined_u:unconfined_r:sepgsql_test_t:s0 tcontext=system_u:object_r:sepgsql_fixed_table_t:s0 tclass=db_column name=c2.d
DELETE FROM c2;
